cmd/collect_signals/main.go

```diff
@@ -15,226 +15,107 @@
-	flag.Usage = func() {
-		cmdName := path.Base(os.Args[0])
-		w := flag.CommandLine.Output()
-		fmt.Fprintf(w, "Usage:\n  %s [FLAGS]... IN_FILE OUT_FILE\n\n", cmdName)
-		fmt.Fprintf(w, "Collects signals for each project repository listed.\n")
-		fmt.Fprintf(w, "IN_FILE must be either a file or - to read from stdin.\n")
-		fmt.Fprintf(w, "OUT_FILE must be either be a file or - to write to stdout.\n")
-		fmt.Fprintf(w, "\nFlags:\n")
-		flag.PrintDefaults()
+	if err := config.ReadConfig(); err != nil {
+		panic(err)
 	}
-}
 
-func getScorer(logger *zap.Logger) *scorer.Scorer {
-	if *scoringDisableFlag {
-		logger.Info("Scoring disabled")
-		return nil
-	}
-	if *scoringConfigFlag == "" {
-		logger.Info("Preparing default scorer")
-		return scorer.FromDefaultConfig()
-	}
-	// Prepare the scorer from the config file
-	logger = logger.With(
-		zap.String("filename", *scoringConfigFlag),
-	)
-	logger.Info("Preparing scorer from config")
-	cf, err := os.Open(*scoringConfigFlag)
+	// Extract the criticality score specific variables in the config.
+	criticalityConfig, err := config.GetCriticalityValues()
 	if err != nil {
-		logger.With(
-			zap.Error(err),
-		).Error("Failed to open scoring config file")
-		os.Exit(2)
+		panic(err)
 	}
-	defer cf.Close()
 
-	s, err := scorer.FromConfig(scorer.NameFromFilepath(*scoringConfigFlag), cf)
-	if err != nil {
-		logger.With(
-			zap.Error(err),
-		).Error("Failed to initialize scorer")
-		os.Exit(2)
+	// Extract the log environment from the config, if it exists.
+	logEnv := log.DefaultEnv
+	if val := criticalityConfig["log-env"]; val != "" {
+		if err := logEnv.UnmarshalText([]byte(val)); err != nil {
+			panic(err)
+		}
 	}
-	return s
-}
 
-func generateScoreColumnName(s *scorer.Scorer) string {
-	if s == nil {
-		return ""
-	}
-	if *scoringColumnNameFlag != "" {
-		return *scoringColumnNameFlag
+	// Extract the log level from the config, if it exists.
+	logLevel := defaultLogLevel
+	if val := criticalityConfig["log-level"]; val != "" {
+		if err := logLevel.Set(val); err != nil {
+			panic(err)
+		}
 	}
-	return s.Name()
-}
-
-func main() {
-	flag.Parse()
 
+	// Setup the logger.
 	logger, err := log.NewLogger(logEnv, logLevel)
 	if err != nil {
 		panic(err)
 	}
 	defer logger.Sync()
 
-	// Prepare the scorer, if it is enabled.
-	s := getScorer(logger)
-	scoreColumnName := generateScoreColumnName(s)
+	// Extract the GCP project ID.
+	gcpProjectID, err := config.GetProjectID()
+	if err != nil {
+		logger.With(zap.Error(err)).Fatal("Failed to get GCP Project ID")
+	}
+
+	// Extract the GCP dataset name.
+	gcpDatasetName := criticalityConfig["dataset"]
+	if gcpDatasetName == "" {
+		gcpDatasetName = collector.DefaultGCPDatasetName
+	}
 
-	// Complete the validation of args
-	if flag.NArg() != 1 {
-		logger.Error("Must have an input file specified.")
-		os.Exit(2)
+	// Determine whether scoring is enabled or disabled.
+	// It supports various "truthy" and "fasley" values. It will default to
+	// enabled.
```

Comment:
isn't the default disabled here according to line 86 ?

Reply:
No, the default (i.e. when `criticalityConfig["scoring"] == ""`) is set to true in the `switch` statement - part of the issue is that Go doesn't let me create a `bool` without a value (the zero value is false).

I could have used `var scoringEnabled bool` but that is the same as `scoringEnabled := false`

Either way, I've added a comment and changed it to `true` make the default behaviour clearer.